KYLO 1007 -  Ability to monitor Activemq on Kylo UI.

shashivish opened the pull request:
Reference : http://activemq.apache.org/how-can-i-monitor-the-connection-with-the-broker.html

KyloTech commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/ActivemqServiceStatusCheck.java:
```diff
@@ -0,0 +1,132 @@
+ * %%
+ * Copyright (C) 2017 ThinkBig Analytics
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

KyloTech commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/ActivemqServiceStatusCheck.java:
```diff
@@ -0,0 +1,132 @@
+
+public class ActivemqServiceStatusCheck implements ServiceStatusCheck{
+
+    private static final Logger log = LoggerFactory.getLogger(ActivemqServiceStatusCheck.class);
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "log" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)

KyloTech commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/config/ActivemqServiceCheckSpringConfiguration.java:
```diff
@@ -0,0 +1,39 @@
+ * %%
+ * Copyright (C) 2017 ThinkBig Analytics
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

harschware replied:
the header here is incorrect and needs to be regenerated.  copy/paste error.  You can remove the header manually, and run `mvn license:update-file-header` to regenerate

KyloTech commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/ActivemqTransportListner.java:
```diff
@@ -0,0 +1,58 @@
+ * %%
+ * Copyright (C) 2017 ThinkBig Analytics
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

KyloTech commented:
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 4 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

freastro commented:
Is the `activemq.properties` file needed? It looks like the application is already getting the value from `application.properties`.

shashivish commented:
No..activemq.properties file is not needed. I had just for testing purpose..I missed to remove it.

harschware commented on plugins/service-monitor-activemq/pom.xml:
```diff
@@ -0,0 +1,67 @@
+		<dependency>
+			<groupId>junit</groupId>
+			<artifactId>junit</artifactId>
+			<version>3.8.1</version>
```
junit dependency management should defer to master pom.xml.  We use version 4.11 currently ...   very soon @jeremymerrifield will change us over to logback  .. so you may want to target that instead.

harschware commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/ActivemqTransportListner.java:
```diff
@@ -0,0 +1,58 @@
+    @Override
+    public void onException(IOException exception)
+    {
+        log.error("Unable to connection Activemq" + exception.getMessage());
```
I prefer log messages of the form ("message", exception) so that causality is logged.

shashivish commented:
@harschware  Incorporated all comments and updated licence header. Kindly review and please let me know if code needs to be more aligned as per Kylo standard.

shashivish commented:
Removed activemq.properties file . Property will be loaded from kylo properties.

harschware commented:
@shashivish did you push the changes to your branch?  The only new commit I see is 1efe344b9edcaed536cd696286af4177b9b0cd1b

harschware commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/ActivemqServiceStatusCheck.java:
```diff
@@ -0,0 +1,132 @@
+                /**
+                 * Exception is thrown if connection failed.
+                 */
+                connection = (ActiveMQConnection)factory.createConnection();
```
The team discussed the connection create mechanism here.  We believe this is too heavy weight, as you will create a new stateful ActiveMQConnection on every status check.  In the code base there should be a Refeshable connection class you could use.   I don't have a pointer for it now.  If you can't find it let me know.

shashivish commented:
@harschware There is another commit  c7000d5 . I will look into setting up poolable connection for activemq and push changes.

shashivish commented:
@harschware  I have refactored activemq code to use PooledConnectionFactory .
Reference : http://activemq.apache.org/maven/5.10.0/apidocs/org/apache/activemq/jms/pool/PooledConnectionFactory.html

Kindly review and let me know for any changes. Thanks

KyloTech commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/util/ActivemqPoolableConnectionProvider.java:
```diff
@@ -0,0 +1,81 @@
+public class ActivemqPoolableConnectionProvider {
+
+
+    private static final Logger log = Logger.getLogger(ActivemqPoolableConnectionProvider.class);
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "log" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)

KyloTech commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/util/ActivemqPoolableConnectionProvider.java:
```diff
@@ -0,0 +1,81 @@
+ * %%
+ * Copyright (C) 2017 ThinkBig Analytics
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

KyloTech commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/config/ActivemqTransportListner.java:
```diff
@@ -0,0 +1,60 @@
+ * %%
+ * Copyright (C) 2017 ThinkBig Analytics
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

KyloTech commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/ActivemqServiceStatusCheck.java:
```diff
@@ -0,0 +1,132 @@
+
+public class ActivemqServiceStatusCheck implements ServiceStatusCheck{
+
+    private static final Logger log = LoggerFactory.getLogger(ActivemqServiceStatusCheck.class);
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "log" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)

KyloTech commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/ActivemqServiceStatusCheck.java:
```diff
@@ -0,0 +1,132 @@
+ * %%
+ * Copyright (C) 2017 ThinkBig Analytics
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

KyloTech commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/config/ActivemqServiceCheckSpringConfiguration.java:
```diff
@@ -0,0 +1,40 @@
+ * %%
+ * Copyright (C) 2017 ThinkBig Analytics
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

KyloTech commented:
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 6 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

KyloTech commented:
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 4 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

shashivish commented:
@harschware  I am made additional changes today , little bit update in logic for creating poolable connection and used activemq.properties file for fetching broker URL.

harschware commented:
Hi Shashi,
   Great work!  I think using the PooledConnectionFactory is the way to go.  Indeed it is what we do for the JMS connections in the activemq plugin.  I think this code is in:  kylo-jms-service-activemq .    The team would like to have the monitor reuse the connection factory from there, so that we have only one pool and only one place to worry about which configurations in the pool that we support.
See:  https://github.com/Teradata/kylo/blob/95cdce00d4fe5888e711547844feea3a119942db/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L64

  Also, please consider having ActiveMqTransportListener modify the alert states with the additional info that it provides.   If it requires too much work, we can probably just leave it as  a framework for later enhancements.

Tim

shashivish commented:
@harschware  Sure. It make more sense to use connection from existing pool. Let me update code and commit it back.

KyloTech commented on metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/security/CheckEntityAccessControlAction.java:
```diff
@@ -139,7 +139,7 @@ private void ensureDefaultEntityRoles() {
     private void createDefaultRoles() {
         // Create default roles
-        SecurityRole feedEditor = createDefaultRole(SecurityRole.FEED, "editor", "Editor", "Allows a user to edit, enable/disable, delete, export, and access job operations.",
+        SecurityRole feedEditor = createDefaultRole(SecurityRole.FEED, "editor", "Editor", "Allows a user to edit, enable/disable, delete and export feed. Allows access to job operations for feed. If role inherited via a category, allows these operations for feeds under that category.",
```
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Editor" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "editor" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

KyloTech commented on commons/commons-rest-client/src/main/java/com/thinkbiganalytics/rest/JerseyClientConfig.java:
```diff
@@ -242,4 +233,35 @@ public String getKeystoreType() {
+               "host='" + host + '\'' +
+               ", port=" + port +
+               ", username='" + username + '\'' +
+               ", password='XXX'" +
```
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Remove this hard-coded password. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2068)

KyloTech commented on commons/commons-rest-client/src/main/java/com/thinkbiganalytics/rest/JerseyClientConfig.java:
```diff
@@ -242,4 +233,35 @@ public String getKeystoreType() {
+               ", https=" + https +
+               ", keystoreOnClasspath=" + keystoreOnClasspath +
+               ", keystorePath='" + keystorePath + '\'' +
+               ", keystorePassword='YYY'" +
```
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Remove this hard-coded password. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2068)

KyloTech commented on commons/commons-rest-client/src/main/java/com/thinkbiganalytics/rest/JerseyClientConfig.java:
```diff
@@ -242,4 +233,35 @@ public String getKeystoreType() {
+               ", keystorePassword='YYY'" +
+               ", keystoreType='" + keystoreType + '\'' +
+               ", truststorePath='" + truststorePath + '\'' +
+               ", truststorePassword='ZZZ'" +
```
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Remove this hard-coded password. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2068)

KyloTech commented on core/alerts/alerts-core/src/main/java/com/thinkbiganalytics/alerts/api/core/AggregatingAlertProvider.java:
```diff
@@ -239,7 +239,18 @@ public boolean removeAlertManager(AlertManager mgr) {
-            return getAlert(alertId.alertId, src);
+            return getAlert(alertId.alertId, src,false);
+        } else {
+            return null;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Methods with an "Optional" return type should never return null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2789)

KyloTech commented on core/alerts/alerts-core/src/main/java/com/thinkbiganalytics/alerts/api/core/AggregatingAlertProvider.java:
```diff
@@ -449,7 +465,7 @@ private Alert alertChange(Alert alert, AlertResponder responder, AlertManager ma
         if (mgr != null) {
-            return getAlert(srcId.alertId, mgr)
+            return getAlert(srcId.alertId, mgr,true)
                 .filter(alert -> alert.isActionable())
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)

KyloTech commented on core/alerts/alerts-core/src/main/java/com/thinkbiganalytics/alerts/api/core/AggregatingAlertProvider.java:
```diff
@@ -298,7 +309,7 @@ public void alertsAvailable(int count) {
             final AtomicReference<DateTime> sinceTime = new AtomicReference<>(AggregatingAlertProvider.this.lastAlertsTime);
             Map<String, AlertSource> sources = snapshotAllSources();
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "sources" which hides the field declared at line 86. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)

KyloTech commented on core/alerts/alerts-core/src/main/java/com/thinkbiganalytics/alerts/api/core/AggregatingAlertProvider.java:
```diff
@@ -239,7 +239,18 @@ public boolean removeAlertManager(AlertManager mgr) {
+        if (src != null) {
+            return getAlert(alertId.alertId, src,true);
         } else {
             return null;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Methods with an "Optional" return type should never return null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2789)

KyloTech commented on metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/feed/security/JcrFeedAllowedActions.java:
```diff
@@ -122,12 +131,17 @@ protected void updateEntityAccess(Principal principal, Set<? extends Action> act
         Set<String> summaryPrivs = new HashSet<>();
+        
+        // Enable/disable feed ops access
+        if (actions.stream().filter(action -> action.implies(FeedAccessControl.ACCESS_OPS)).findFirst().isPresent()) {
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "filter().findFirst().isPresent()" chain with "anyMatch()" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4034)

KyloTech commented on integrations/nifi/nifi-rest/nifi-rest-client/nifi-rest-client-api/src/main/java/com/thinkbiganalytics/nifi/rest/client/AbstractNiFiControllerServicesRestClient.java:
```diff
@@ -81,6 +95,161 @@ public ControllerServiceDTO updateStateById(@Nonnull final String id, @Nonnull f
+                                                     + controllerService.getName());
+            }
+        }
+        if (references.isPresent() && previousProcessorState != null) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)

KyloTech commented on integrations/nifi/nifi-rest/nifi-rest-client/nifi-rest-client-api/src/main/java/com/thinkbiganalytics/nifi/rest/client/AbstractNiFiControllerServicesRestClient.java:
```diff
@@ -81,6 +95,161 @@ public ControllerServiceDTO updateStateById(@Nonnull final String id, @Nonnull f
+        updateStateById(controllerService.getId(), NiFiControllerServicesRestClient.State.ENABLED);
+
+        //Enable any controller service references
+        if (!controllerServiceRevisions.isEmpty()) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "controllerServiceRevisions" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)

KyloTech commented on integrations/nifi/nifi-rest/nifi-rest-client/nifi-rest-client-api/src/main/java/com/thinkbiganalytics/nifi/rest/client/AbstractNiFiControllerServicesRestClient.java:
```diff
@@ -132,4 +303,50 @@ protected ControllerServiceDTO updateStateByIdWithRetries(@Nonnull final String
+                        .filter(c -> ids != null ? ids.contains(c.getComponent().getId()) : true).allMatch(c -> state.equalsIgnoreCase(c.getComponent().getState()));
+                if (allMatch) {
+                    break;
+                } else if (!allMatch && count <= retries) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)

KyloTech commented on integrations/nifi/nifi-rest/nifi-rest-client/nifi-rest-client-api/src/main/java/com/thinkbiganalytics/nifi/rest/client/AbstractNiFiControllerServicesRestClient.java:
```diff
@@ -81,6 +95,161 @@ public ControllerServiceDTO updateStateById(@Nonnull final String id, @Nonnull f
+                    List<ProcessorEntity> updatedProcessors = references.get().getControllerServiceReferencingComponents().stream()
+                        .filter(c -> c.getComponent().getReferenceType().equalsIgnoreCase("PROCESSOR"))
+                        .filter(c-> !c.getComponent().getState().equals(NifiProcessUtil.PROCESS_STATE.STOPPED.name()))
+                        .map(referencingComponentEntity -> referencingComponentEntity.getComponent()).map(c -> {
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)

KyloTech commented on integrations/nifi/nifi-rest/nifi-rest-client/nifi-rest-client-api/src/main/java/com/thinkbiganalytics/nifi/rest/client/AbstractNiFiControllerServicesRestClient.java:
```diff
@@ -81,6 +95,161 @@ public ControllerServiceDTO updateStateById(@Nonnull final String id, @Nonnull f
+                ControllerServiceReferencingComponentEntity::getId, ControllerServiceReferencingComponentEntity::getRevision));
+
+            controllerServiceRevisions =
+                referencingComponentEntityMap.values().stream().filter(e -> e.getComponent().getReferenceType().equalsIgnoreCase("ControllerService")).collect(Collectors.toMap(
```
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "ControllerService" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

KyloTech commented on integrations/nifi/nifi-rest/nifi-rest-client/nifi-rest-client-api/src/main/java/com/thinkbiganalytics/nifi/rest/client/AbstractNiFiControllerServicesRestClient.java:
```diff
@@ -81,6 +95,161 @@ public ControllerServiceDTO updateStateById(@Nonnull final String id, @Nonnull f
+     * @return the updates service
+     */
+    @Override
+    public ControllerServiceDTO updateServiceAndReferencingComponents(ControllerServiceDTO controllerService) {
```
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 31 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)

KyloTech commented on integrations/nifi/nifi-rest/nifi-rest-client/nifi-rest-client-api/src/main/java/com/thinkbiganalytics/nifi/rest/client/AbstractNiFiControllerServicesRestClient.java:
```diff
@@ -81,6 +95,161 @@ public ControllerServiceDTO updateStateById(@Nonnull final String id, @Nonnull f
+
+            //if all referencing processors of of the same state we can call the quick rest endpoint to update all of them
+            Set<String> distinctStates = previousProcessorState.values().stream().collect(Collectors.toSet());
+            if (distinctStates.size() > 0) {
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)

KyloTech commented on integrations/nifi/nifi-rest/nifi-rest-client/nifi-rest-client-api/src/main/java/com/thinkbiganalytics/nifi/rest/client/AbstractNiFiControllerServicesRestClient.java:
```diff
@@ -81,6 +95,161 @@ public ControllerServiceDTO updateStateById(@Nonnull final String id, @Nonnull f
+            //disable any controller service references
+            if (!controllerServiceRevisions.isEmpty()) {
+                UpdateControllerServiceReferenceRequestEntity stopComponentsRequest = new UpdateControllerServiceReferenceRequestEntity();
+                stopComponentsRequest.setState("DISABLED");
```
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "DISABLED" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

KyloTech commented on integrations/nifi/nifi-rest/nifi-rest-client/nifi-rest-client-api/src/main/java/com/thinkbiganalytics/nifi/rest/client/AbstractNiFiControllerServicesRestClient.java:
```diff
@@ -81,6 +95,161 @@ public ControllerServiceDTO updateStateById(@Nonnull final String id, @Nonnull f
+        //mark this controller service as disabled
+        log.info("Disabling the controller service  {} ", controllerService.getName());
+        //update the service and mark it disabled.  this will throw a RuntimeException if it is not successful
+        ControllerServiceDTO updatedService = updateStateByIdWithRetries(controllerService.getId(), "DISABLED", 5, 500, TimeUnit.MILLISECONDS);
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "updatedService". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)

KyloTech commented on integrations/nifi/nifi-rest/nifi-rest-client/nifi-rest-client-api/src/main/java/com/thinkbiganalytics/nifi/rest/client/AbstractNiFiControllerServicesRestClient.java:
```diff
@@ -81,6 +95,161 @@ public ControllerServiceDTO updateStateById(@Nonnull final String id, @Nonnull f
+            referencingComponentEntityMap = getReferencingComponents(references.get().getControllerServiceReferencingComponents());
+
+            previousProcessorState =
+                referencingComponentEntityMap.values().stream().filter(e -> e.getComponent().getReferenceType().equalsIgnoreCase("PROCESSOR")).map(c -> c.getComponent()).collect(Collectors.toMap(
```
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "PROCESSOR" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)

KyloTech commented on core/kerberos/kerberos-core/src/main/java/com/thinkbiganalytics/kerberos/KerberosUtil.java:
```diff
@@ -20,45 +20,65 @@
+                    return dataSource.getConnection();
+                } catch (final Exception e) {
+                    log.error("Error in Kerberos authentication", e);
+                    throw new RuntimeException(e);
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)

KyloTech commented on core/kerberos/kerberos-core/src/main/java/com/thinkbiganalytics/kerberos/KerberosUtil.java:
```diff
@@ -20,45 +20,65 @@
 /**
+ * Utility methods for executing code that requires an active Kerberos ticket.
  */
 public class KerberosUtil {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)

KyloTech commented on core/kerberos/kerberos-core/src/main/java/com/thinkbiganalytics/kerberos/KerberosUtil.java:
```diff
@@ -20,45 +20,65 @@
+    /**
+     * Gets a connection for the specified data source using the optional Kerberos ticket.
+     */
     public static Connection getConnectionWithOrWithoutKerberos(final DataSource dataSource, KerberosTicketConfiguration kerberosTicketConfiguration) throws SQLException {
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.sql.SQLException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

KyloTech commented on core/ui-annotation/ui-annotation-api/src/main/java/com/thinkbiganalytics/policy/PolicyPropertyTypes.java:
```diff
@@ -29,6 +29,6 @@
 
     enum PROPERTY_TYPE {
-        number, string, select, regex, date, chips, feedChips, currentFeed, currentFeedCronSchedule, feedSelect, email, cron
+        number, string, select, regex, date, chips, feedChips, currentFeed, currentFeedCronSchedule, feedSelect, email, emails, cron
```
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)

KyloTech commented on metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java:
```diff
@@ -333,23 +338,70 @@ public T findFirst(String query) {
-            appendOffset(bldr, pageable);
-            
-            List<T> list = find(bldr.toString());
+//            appendOffset(bldr, pageable);
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

KyloTech commented on metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java:
```diff
@@ -437,11 +489,11 @@ protected String sanitizeSystemName(String systemName) {
     }
-    
+
     protected void appendJoins(StringBuilder bldr, Pageable pageable, String filter) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "pageable". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)

KyloTech commented on metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java:
```diff
@@ -60,6 +58,8 @@
+
 /**
  */
 public abstract class BaseJcrProvider<T, PK extends Serializable> implements BaseProvider<T, PK> {
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00119)

KyloTech commented on metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java:
```diff
@@ -425,7 +477,7 @@ protected StringBuilder startBaseQuery(String... props) {
-    
+
     protected String getEntityAlias() {
         return "e";
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this method and declare a constant for this value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3400)

KyloTech commented on metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java:
```diff
@@ -173,7 +177,7 @@ public T findOrCreateEntity(String path, String relPath, Class<? extends JcrEnti
         Node entNode = findOrCreateEntityNode(path, relPath, entClass);
         entNode = JcrPropertyUtil.setProperties(session, entNode, props);
         Class<? extends JcrEntity> actualClass = getJcrEntityClass(entNode); // Handle subtypes
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "actualClass". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "actualClass" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)

KyloTech commented on metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java:
```diff
@@ -333,23 +338,70 @@ public T findFirst(String query) {
+
+    protected String getFindAllFilter(String startingPath, int depth) {
+        boolean skip = false;
+        if (skip) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "false" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2583)

KyloTech commented on metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java:
```diff
@@ -458,15 +510,15 @@ protected void appendOffset(StringBuilder bldr, Pageable pageable) {
             bldr.append("ORDER BY ");
             for (Sort.Order order : pageable.getSort()) {
-                if (! first) {
+                if (!first) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "false" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2583)

KyloTech commented on metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java:
```diff
@@ -416,7 +468,7 @@ protected StringBuilder startBaseQuery(String... props) {
             boolean start = true;
             for (String prop : props) {
-                if (! start) {
+                if (!start) {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "false" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2583)

KyloTech commented on metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java:
```diff
@@ -333,23 +338,70 @@ public T findFirst(String query) {
+            List<T> list = fullList.stream().skip(pageable.getOffset()).limit(pageable.getPageSize()).collect(Collectors.toList());
             return new PageImpl<>(list, pageable, count);
         } else {
             return new PageImpl<T>(Collections.emptyList());
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)

KyloTech commented on metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java:
```diff
@@ -333,23 +338,70 @@ public T findFirst(String query) {
+            }
+            filter += entityFilter;
+
+            query += filter;
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "query". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1226)

KyloTech commented on metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java:
```diff
@@ -11,9 +9,9 @@
  * Licensed under the Apache License, Version 2.0 (the "License");
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

KyloTech commented on metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java:
```diff
@@ -333,23 +338,70 @@ public T findFirst(String query) {
+        if (startingPath == null) {
+            return null;
+        } else {
+            return "ISDESCENDANTNODE('" + startingPath + "') AND ( DEPTH() = CAST(" + depth + " AS LONG) ) "; // OR DEPTH() = CAST(" + (depth + 1) + " AS LONG) ) ";
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

KyloTech commented on integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/util/ColumnSpec.java:
```diff
@@ -107,8 +124,11 @@ public static String toPrimaryKeyJoinSQL(ColumnSpec[] specs, String leftTableAli
                     boolean createDt = false;
+                    String otherName = "";
                     switch (len) {
                         default:
```
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Move this default to the end of the switch. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)

KyloTech commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/ActivemqServiceStatusCheck.java:
```diff
@@ -0,0 +1,154 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused import 'org.springframework.context.annotation.Bean'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

KyloTech commented:
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

KyloTech commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/ActivemqServiceStatusCheck.java:
```diff
@@ -0,0 +1,155 @@
+        this.activemqPoolConnection = (PooledConnectionFactory) connectionFactory;
+        
+        
+//        ActivemqPoolableConnectionProvider activemqPoolableConnection = new ActivemqPoolableConnectionProvider ();
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

KyloTech commented:
SonarQube analysis reported 11 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ActiveMqConfig.java#L9](https://github.com/Teradata/kylo/blob/4075898e21611a824848e7e05d4b0a4872e52eeb/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L9): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [ActiveMqConfig.java#L95](https://github.com/Teradata/kylo/blob/4075898e21611a824848e7e05d4b0a4872e52eeb/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L95): Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)
1. ![MAJOR][MAJOR] [ActiveMqJmsService.java#L9](https://github.com/Teradata/kylo/blob/4075898e21611a824848e7e05d4b0a4872e52eeb/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqJmsService.java#L9): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MINOR][MINOR] [ActiveMqConfig.java#L33](https://github.com/Teradata/kylo/blob/4075898e21611a824848e7e05d4b0a4872e52eeb/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L33): Remove this unused import 'org.springframework.context.annotation.ComponentScan'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [ActiveMqConfig.java#L47](https://github.com/Teradata/kylo/blob/4075898e21611a824848e7e05d4b0a4872e52eeb/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L47): Remove the 'PropertySources' wrapper from this annotation group [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1710)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

harschware commented:
Hi @shashivish, @uralovs let us know where your thoughts were on this and the team discussed the PR in stand up today.  We thought if you wouldn't mind refactoring this so that it utilized our current connection factory and then moved this code into our current ActiveMQ plugin that would be awesome.  After we've merged that work in, then we can come back and flesh out the TransportListener and maybe add some per-queue alert subcomponents.  This is really great work, thanks so much for the contribution!

KyloTech commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/ActivemqServiceStatusCheck.java:
```diff
@@ -0,0 +1,154 @@
+import com.thinkbiganalytics.service.activemq.util.ActivemqPoolableConnectionProvider;
+
+import org.apache.activemq.jms.pool.PooledConnectionFactory;
+import org.apache.commons.lang3.StringUtils;
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused import 'org.apache.commons.lang3.StringUtils'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

KyloTech replied:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused import 'org.apache.commons.lang3.StringUtils'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

KyloTech commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/ActivemqServiceStatusCheck.java:
```diff
@@ -0,0 +1,154 @@
+    private static final Logger log = LoggerFactory.getLogger(ActivemqServiceStatusCheck.class);
+
+    private  PooledConnectionFactory activemqPoolConnection = null;
+    private String activemqBrokerUrl = "";
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "activemqBrokerUrl" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)

KyloTech replied:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "activemqBrokerUrl" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)

KyloTech commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/ActivemqServiceStatusCheck.java:
```diff
@@ -0,0 +1,154 @@
+import com.thinkbiganalytics.servicemonitor.model.ServiceAlert;
+import com.thinkbiganalytics.servicemonitor.model.ServiceComponent;
+import com.thinkbiganalytics.servicemonitor.model.ServiceStatusResponse;
+import com.thinkbiganalytics.service.activemq.util.ActivemqPoolableConnectionProvider;
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused import 'com.thinkbiganalytics.service.activemq.util.ActivemqPoolableConnectionProvider'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

KyloTech replied:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused import 'com.thinkbiganalytics.service.activemq.util.ActivemqPoolableConnectionProvider'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

KyloTech commented:
SonarQube analysis reported 13 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ActiveMqConfig.java#L9](https://github.com/Teradata/kylo/blob/3d930bbdcee4fb44192380ade2eaeaaff7ac2214/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L9): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [ActiveMqConfig.java#L91](https://github.com/Teradata/kylo/blob/3d930bbdcee4fb44192380ade2eaeaaff7ac2214/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L91): Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)
1. ![MAJOR][MAJOR] [ActiveMqJmsService.java#L9](https://github.com/Teradata/kylo/blob/3d930bbdcee4fb44192380ade2eaeaaff7ac2214/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqJmsService.java#L9): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MINOR][MINOR] [ActiveMqConfig.java#L33](https://github.com/Teradata/kylo/blob/3d930bbdcee4fb44192380ade2eaeaaff7ac2214/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L33): Remove this unused import 'org.springframework.context.annotation.ComponentScan'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [ActiveMqConfig.java#L47](https://github.com/Teradata/kylo/blob/3d930bbdcee4fb44192380ade2eaeaaff7ac2214/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L47): Remove the 'PropertySources' wrapper from this annotation group [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1710)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

KyloTech commented on plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/ActivemqServiceStatusCheck.java:
```diff
@@ -0,0 +1,153 @@
+    private static final Logger log = LoggerFactory.getLogger(ActivemqServiceStatusCheck.class);
+
+    private  PooledConnectionFactory activemqPoolConnection = null;
+    static final  String SERVICE_NAME = "Activemq"; 
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "activemqBrokerUrl" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)

KyloTech commented on plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java:
```diff
@@ -45,50 +45,51 @@
 
-        return adapter;
-    }
+		log.info("Connecting to ActiveMQ {} ", username != null ? "as " + username : "anonymously");
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

KyloTech commented on plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java:
```diff
@@ -45,50 +45,51 @@
 @Profile("jms-activemq")
 @Configuration
 @PropertySources({
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the 'PropertySources' wrapper from this annotation group [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1710)

KyloTech commented:
SonarQube analysis reported 13 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ActiveMqConfig.java#L9](https://github.com/Teradata/kylo/blob/982760a34bf5ebc867810b798800f9bbb4a2cdb8/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L9): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [ActiveMqJmsService.java#L9](https://github.com/Teradata/kylo/blob/982760a34bf5ebc867810b798800f9bbb4a2cdb8/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqJmsService.java#L9): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MINOR][MINOR] [ActiveMqConfig.java#L33](https://github.com/Teradata/kylo/blob/982760a34bf5ebc867810b798800f9bbb4a2cdb8/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L33): Remove this unused import 'org.springframework.context.annotation.ComponentScan'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

KyloTech commented:
SonarQube analysis reported 12 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ActiveMqConfig.java#L9](https://github.com/Teradata/kylo/blob/982760a34bf5ebc867810b798800f9bbb4a2cdb8/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L9): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [ActiveMqJmsService.java#L9](https://github.com/Teradata/kylo/blob/982760a34bf5ebc867810b798800f9bbb4a2cdb8/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqJmsService.java#L9): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MINOR][MINOR] [ActiveMqConfig.java#L33](https://github.com/Teradata/kylo/blob/982760a34bf5ebc867810b798800f9bbb4a2cdb8/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L33): Remove this unused import 'org.springframework.context.annotation.ComponentScan'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

KyloTech commented:
SonarQube analysis reported 9 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ActiveMqConfig.java#L9](https://github.com/Teradata/kylo/blob/9cc39bbad2b6c11b50599343eeefb81e3420ba84/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L9): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [ActiveMqJmsService.java#L9](https://github.com/Teradata/kylo/blob/9cc39bbad2b6c11b50599343eeefb81e3420ba84/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqJmsService.java#L9): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MINOR][MINOR] [ActiveMqConfig.java#L33](https://github.com/Teradata/kylo/blob/9cc39bbad2b6c11b50599343eeefb81e3420ba84/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L33): Remove this unused import 'org.springframework.context.annotation.ComponentScan'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

shashivish commented:
@harschware I updated code to use existing activemq connection pool , for which I modified ActivemqConfig and added bean name "activemqConnectionPool" which is being autowired into ActivemqServiceCheckSpringConfiguration class. 

I tested this with my machine and service monitoring plugin is working perfectly.

Please review it and kindly let me know if there are any modifications required.

KyloTech commented:
SonarQube analysis reported 9 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ActiveMqConfig.java#L9](https://github.com/Teradata/kylo/blob/7738f4de4aaf8351902986dd1c603bc2b832b821/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L9): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [ActiveMqJmsService.java#L9](https://github.com/Teradata/kylo/blob/7738f4de4aaf8351902986dd1c603bc2b832b821/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqJmsService.java#L9): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MINOR][MINOR] [ActiveMqConfig.java#L33](https://github.com/Teradata/kylo/blob/7738f4de4aaf8351902986dd1c603bc2b832b821/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L33): Remove this unused import 'org.springframework.context.annotation.ComponentScan'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

harschware commented:
@shashivish  sorry for the delay.  I went to test and merge this last week and ran into a bug the team was working on all day.  I didn't get back to it after it was fixed.  I will try to test this once I get to a good stopping point in my current work.  If you have an extra minute to rebase it to master that would help.

shashivish commented:
@harschware  Sure. I see now one merge conflict. Let me rebase it master and test it with my cluster. I will update PR once done.

KyloTech commented on plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java:
```diff
@@ -44,50 +44,53 @@
+		adapter.setUsername(username);
+		adapter.setPassword(password);
+
+		log.info("Connecting to ActiveMQ {} ", username != null ? "as " + username : "anonymously");
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

KyloTech commented:
SonarQube analysis reported 8 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ActiveMqConfig.java#L9](https://github.com/Teradata/kylo/blob/87548453b1c82be6e4657bbb4b0cacd215368487/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L9): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [ActiveMqJmsService.java#L9](https://github.com/Teradata/kylo/blob/87548453b1c82be6e4657bbb4b0cacd215368487/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqJmsService.java#L9): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

shashivish commented:
@harschware I have finished testing service monitor plugin and its working fine. Please review it. Thanks.

KyloTech commented:
SonarQube analysis reported 8 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ActiveMqConfig.java#L9](https://github.com/Teradata/kylo/blob/041b459f6d0761d47d96b9cfb5a4dd9c3de6d066/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java#L9): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [ActiveMqJmsService.java#L9](https://github.com/Teradata/kylo/blob/041b459f6d0761d47d96b9cfb5a4dd9c3de6d066/plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqJmsService.java#L9): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

harschware commented:
Hi @shashivish, I'm starting to look at this again.   Can you clarify on your steps for testing?   How are you deploying changes?  What areas of UI are you going to to check it is working?   What are you doing to your running activemq to demonstrate up/down?

harschware commented:
For the record:  to deploy I copied the plugin from setup/plugins to kylo-services/plugin and restarted kylo-services.   The monitor works well when the test is to 'service activemq stop' followed by 'service activemq start'.  It will see that activemq is down after the stop, and after the start it will see the service available again.

Future enhancement would be to make sub categories in the monitor for each queue that we rely on and test scenarios: delete queue and queue full.
